src/sentry/integrations/slack/notifications.py

```diff
@@ -54,11 +53,7 @@ def _get_attachments(
     context = get_context(notification, recipient, shared_context, extra_context)
     cls = get_message_builder(notification.message_builder)
     attachments = cls(notification, context, recipient).build()
-    if isinstance(attachments, list) or features.has(
-        "organizations:slack-block-kit", notification.organization
-    ):
-        return attachments
-    return [attachments]
+    return attachments
```

Comment:
we we still need the type check logic?

Reply:
this one i'm not sure cc @ceorourke @isabellaenriquez

Reply:
Looking at the git blame I think we do, it looks like it was added [here](https://github.com/getsentry/sentry/commit/c7439de6de3535c3d2000c226f79fac34a79daa4)

Reply:
the logic was changed here for block kit

<img width="750" alt="Screenshot 2024-04-15 at 15 54 24" src="https://github.com/getsentry/sentry/assets/70817427/7f633698-d9ac-47ed-bc11-ff1b574376fa">

if block kit is default True, the conditional is always true and we never return `[attachments]`

Reply:
I think we'd have to update the type hint for the return type too then, would updating that help identify any problems?

Reply:
I think we can remove the list, looks like it was only needed for non-block kit flows? The downstream of this is only used in `_notify_recipient`, which has a check for the FF and logic accordingly

Reply:
We'll have to update the param type in `_notify_recipient` so that the change is consistent/safe, as we don't need the list anymore

Reply:
Could we do one layer at a time? Or is it a cascade effect?

Reply:
this function calls `build()` and it could be for any of the slack message builder classes

Reply:
the build() is `SlackAttachment | SlackBlock`

Reply:
actually you're right, we can just update the downstream types

Reply:
We could also make `notification: BaseNotification,` into the Slack notification builder, since this is relegated to Slack flows, and would make the typing for the build return type easier as well